otx/core/data/anomaly_dataset_adapter.py

```diff
@@ -0,0 +1,220 @@
+from otx.api.entities.label_schema import LabelSchemaEntity
+from otx.api.entities.scored_label import ScoredLabel
+from otx.api.entities.shapes.rectangle import Rectangle
+from otx.api.entities.subset import Subset
+from otx.api.utils.segmentation_utils import create_annotation_from_segmentation_map
+from otx.core.data.base_dataset_adapter import BaseDatasetAdapter
+
+
+class AnomalyBaseDatasetAdapter(BaseDatasetAdapter):
+    """BaseDataset Adpater for Anomaly tasks inherited from BaseDatasetAdapter."""
+
+    def import_dataset(
+        self,
+        train_data_roots: str = None,
+        val_data_roots: str = None,
+        test_data_roots: str = None,
+        unlabeled_data_roots: str = None,
+    ) -> Dict[Subset, DatumaroDataset]:
+        """Import dataset by using Datumaro.import_from() method.
+
+        Args:
+            train_data_roots (str): Path for training data
+            val_data_roots (str): Path for validation data
+            test_data_roots (str): Path for test data
+            unlabeled_data_roots (str): Path for unlabeled data
+
+        Returns:
+            DatumaroDataset: Datumaro Dataset
+        """
+        # Construct dataset for training, validation, unlabeled
+        # TODO: currently, only MVTec dataset format can be used
+        self.dataset = {}
+        self.dataset[Subset.TRAINING] = DatumaroDataset.import_from(train_data_roots, format="image_dir")
+        if val_data_roots:
+            self.dataset[Subset.VALIDATION] = DatumaroDataset.import_from(val_data_roots, format="image_dir")
+        return self.dataset
+
+    def _prepare_anomaly_label_information(self) -> List[LabelEntity]:
+        """Prepare LabelEntity List."""
+        normal_label = LabelEntity(id=ID(0), name="Normal", domain=self.domain)
+        abnormal_label = LabelEntity(
+            id=ID(1),
+            name="Anomalous",
+            domain=self.domain,
+            is_anomalous=True,
+        )
+        return [normal_label, abnormal_label]
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Anomalytasks."""
+        raise NotImplementedError
+
+
+class AnomalyClassificationDatasetAdapter(AnomalyBaseDatasetAdapter):
+    """Anomaly classification adapter inherited from AnomalyBaseDatasetAdapter."""
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Anomaly classification."""
+        normal_label, abnormal_label = self._prepare_anomaly_label_information()
+        label_schema = self._generate_default_label_schema([normal_label, abnormal_label])
+
+        # Prepare
+        dataset_items = []
+        for subset, subset_data in datumaro_dataset.items():
+            for _, datumaro_items in subset_data.subsets().items():
+                for datumaro_item in datumaro_items:
```

Comment:
It would be nice if this part (for statement) was also subtracted as a function.